File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer, Integer> dic = new HashMap<>();
        for (int i: nums) {
            dic.put(i, dic.getOrDefault(i,0) + 1);
        }

        // System.out.println(dic);

        for (int x : dic.values()) {
            // System.out.println(x);
            if (x > 1) return true;
        }

        return false;
        
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        
        while l<r:
            mid = l + (r-l)//2
            # print(mid)

            if nums[mid] > nums[r]:
                l = mid + 1
                # print("nums[l] = ", nums[l])
            else:
                r = mid
                # print("nums[r] = ", nums[r])
        return nums[l]
        